imgproc.go

```diff
@@ -1343,6 +1413,51 @@ func FillPoly(img *Mat, pts [][]image.Point, c color.RGBA) {
 	C.FillPoly(img.p, cPoints, sColor)
 }
 
+// FillPolyWithParams fills the area bounded by one or more polygons.
+//
+// For more information, see:
+// https://docs.opencv.org/master/d6/d6e/group__imgproc__draw.html#gaf30888828337aa4c6b56782b5dfbd4b7
+func FillPolyWithParams(img *Mat, pts [][]image.Point, c color.RGBA, lineType LineType, shift int, offset image.Point) {
+	points := make([]C.struct_Points, len(pts))
+	offsetP := C.struct_Point{
+		x: C.int(offset.X),
+		y: C.int(offset.Y),
+	}
+
+	for i, pt := range pts {
+		p := (*C.struct_Point)(C.malloc(C.size_t(C.sizeof_struct_Point * len(pt))))
+		defer C.free(unsafe.Pointer(p))
```

Comment:
I do not think you want to use `defer` here. Instead I suggest explicitly freeing `p` at the end of each loop iteration.

Reply:
Make sense. I just used existing working code for `WithParams` functions, so old code needs to be refactored as well.

Reply:
No doubt an audit of the code will reveal a few more of these to be fixed.